docs/okeps/okep-5233-pre-assigned-pod-net-id.md

```diff
@@ -0,0 +1,344 @@
+* 'v1.multus-cni.io/default-network' cannot be changed after the pod was created.
+* It should be possible to configure the pods MAC or the IP address without configuring the other.
+* Detect MAC and IP address conflicts between the requested addresses for a newly created pods and the addresses that
+are already allocated in the network.
+* After configuring custom default gateway and management addresses on a Layer2 UDN the previous default
+IPs can be consumed by workloads(e.g. for 10.0.0.0/16 network create pods with 10.0.0.1 and 10.0.0.2 addresses).
+* Modifying the  default gateway and management addresses on a Layer2 UDN should not be possible after the network
+was created.
+
+The scenarios mentioned above have to cover both IPv4 and IPv6 IP families.
+
+
+### Documentation Details
+
+## Risks, Known Limitations and Mitigations
+
+* Modifying the 'v1.multus-cni.io/default-network' value after the pod was created could have unpredictable
+consequences.
+To mitigate this introduce a Validating Admission Policy described in [Implementation Details](#implementation-details).
+
+* By allowing users to specify the IP and MAC addresses for the pods there is a risk of conflicts.
+To mitigate this OVN-Kubernetes will check that the requested addresses are not currently used in the UDN.
+There is still a risk that the user picks an address that's consumed by something outside of the UDN but that's beyond
+what OVN-Kubernetes controls and can check.
+
+* The dynamic, per-node subnet allocation in Layer3 UDNs, where each node has a unique default gateway and
+management IP, makes user-specified UDN gateway/management IPs and static pod IP/MAC assignments very complex. This
+enhancement will not support Layer3 UDNs.
+
+* BGP support today is limited to cluster UDNs, to ensure a non-NATed traffic for pods with pre-configured addresses
+the user has to use a cluster UDN to configure the network. This is a limitation unrelated to this enhancement
+and it is possible it will be solved in the future.
+
+* By consuming the 'v1.multus-cni.io/default-network' annotation for altering the primary UDNs pod configuration the
+user won't be able to use it for configuring the cluster default network attachment. This is acceptable as there is
+currently no support for modifying the cluster default network through this annotation while using primary UDNs.
+If there is a requirement in the future another mechanism can be considered.
+
+## OVN Kubernetes Version Skew
+
+## Alternatives
+
+Instead of the [Pod network identity](#pod-network-identity) approach, we could expand the
+IPAMClaim API. It currently lacks IP request capabilities, and using IPAMClaim for MAC addresses
+is confusing. Introducing a new API would mean deprecating the IPAMClaim, while managing
+upgrades and supporting both solutions for a period of time. This requires significant effort, which
+is not feasible at this time.
```

Comment:
This alternative is not part of ovn-kubernetes, the entity that reads IPAMClaim will still be converting it into default-network annotation, ovnk should just understand that.

Reply:
OVN-Kubernetes sets the status on the IPAMClaim already and if the IPAMClaim changes the spec I don't see why ovn-k should rely on the default-network annotation instead.
This would be a really weird behavior for OVN-K to read some annotation to set the status while ignoring the spec, it could lead to some bad scenarios where the spec doesn't match the annotation.